### DISCUSSION TITLE
Q_nominal?

### DISCUSSION THREAD

**Discussion by gitfeldhase** [oemof/DHNx#114] [Q&A]
Hello everyone,

i am facing another question.
I was looking for the constraints for the optimisation and therefore checking out the HeatPipeline class.
I saw the variable "nominal capacity of heating pipeline".
However, this parameter is not defined in the invest data for the pipes (invest_data/network/pipes.csv).

Therefore I assume that in case the **nominal heat flow** Q_nominal is not defined, it will use Q_out of the adjacent pipe to calculate the losses.

`.. math:: \dot{Q}_{loss}(t) = f_{loss}(t) \cdot l \cdot \dot{Q}_{nominal}`

Is that correct?

This is where I think the nominal value (Q_nominal) is set:
`class HeatPipeline(Transformer):`
[...]

    def _set_nominal_value(self):
        i = list(self.inputs.keys())[0]
        o = list(self.outputs.keys())[0]
        if self.outputs[o].nominal_value is not None:
            self.inputs[i].nominal_value = \
                self.outputs[o].nominal_value
        elif self.inputs[i].nominal_value is not None:
            self.outputs[o].nominal_value = \
                self.inputs[i].nominal_value

Best regards and thanks for the support.

**Comment by joroeder** [comment]
I am currently out of office. In short, there are two constraint blocks, if you are doing investment optimization, the nomial_value is none, because it is an decision variable. Think of extending an existing network, therefore you can also define heatpipelines with given nominal_value.

**Reply by clarapineau** [reply]
Hello, I am facing the same question regarding the extension of an existing network. Is it correct that for existing pipes I have to specify their capacity and there is no way to just specify a pipe type + length and let DHNx calculate the capacity?
Thanks for your work and kind regards.
